http/http2/src/main/java/io/helidon/http/http2/Http2ConnectionWriter.java

```diff
@@ -143,24 +142,14 @@ public int writeHeaders(Http2Headers headers,
                             Http2Flag.HeaderFlags flags,
                             Http2FrameData dataFrame,
                             FlowControl.Outbound flowControl) {
-        // this is executing in the thread of the stream
-        // we must enforce parallelism of exactly 1, to make sure the dynamic table is updated
-        // and then immediately written
-
-        lock();
-        try {
-            int bytesWritten = 0;
-
-            bytesWritten += writeHeaders(headers, streamId, flags, flowControl);
-
-            writeData(dataFrame, flowControl);
-            bytesWritten += Http2FrameHeader.LENGTH;
-            bytesWritten += dataFrame.header().length();
-
-            return bytesWritten;
-        } finally {
-            streamLock.unlock();
-        }
+        // Executed on stream thread
```

Comment:
Could you explain why the mutual exclusion is no longer needed here?

Reply:
Because method is executed only from a single stream thread only, lock was actually causing a deadlock when stream was expecting flow-control window update(wating for update blocks stream thread). Headers needs to be followed by data only from a single stream point of view, other frames from other streams needs and can be writable between those two.